web/policies/preview_policy.ex

```diff
@@ -1,5 +1,7 @@
 defmodule CodeCorps.PreviewPolicy do
   alias CodeCorps.User
 
-  def create?(%User{} = _user), do: true
+  alias Ecto.Changeset
+
+  def create?(%User{} = user, %Changeset{changes: %{user: %{data: creator}}}), do: user.id == creator.id
```

Comment:
The preview uses the `current_user` and the changeset assigns that user. We're checking that the creator of the preview is the user trying to create it.